### PR TITLE
Move benchmarks to JMH

### PR DESCRIPTION
**Pull request by ianoc**
Moves the benchmark suite to jmh

### CODE REVIEW

File: algebird-benchmark/src/main/scala/com/twitter/algebird/benchmark/HllBatchCreateBenchmark.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    }
+  }
+}
+
+class HllBatchCreateBenchmark {
+  import HllBatchCreateBenchmark._
+
+  @Benchmark
+  def timeBatchCreate(state: HLLState, bh: Blackhole) = {
+    bh.consume(state.hllMonoid.batchCreate(state.set)(byteEncoderFn))
```
Comment (jnievelt) [inline algebird-benchmark/src/main/scala/com/twitter/algebird/benchmark/HllBatchCreateBenchmark.scala]: bh doesn't seem to be needed here

File: algebird-benchmark/src/main/scala/com/twitter/algebird/benchmark/AsyncSummerBenchmark.scala [hunk 2]
```diff
@@ -0,0 +1,156 @@
+package com.twitter.algebird.benchmark
+
+import java.lang.Math._
+import java.util.concurrent.Executors
+import java.util.concurrent.atomic.AtomicLong
+
+import java.util.concurrent.TimeUnit
+import org.openjdk.jmh.annotations._
+import org.openjdk.jmh.infra.Blackhole
```
Comment (jnievelt) [inline algebird-benchmark/src/main/scala/com/twitter/algebird/benchmark/AsyncSummerBenchmark.scala]: This import isn't used in a few places

**Comment by johnynek** [comment]
The README needs to be updated. We moved it over, but didn't explain how to run the new benchmarks.

I don't know so I'd enjoy the docs myself.

**Comment by ianoc** [comment]
Updated the docs including example of running

**Comment by ianoc** [comment]
Script i've been using with this to generate matrices of results -- uses two repos currently that i have setup but could do branches. Doubt its worth adding to the repo, but if anyone is interested.

```bash
set +e
DATE_HR=`date "+%Y%m%d%H%M"`
OUTPUT_FOLDER=~/benchmark_runs/$DATE_HR
mkdir -p $OUTPUT_FOLDER

JAVA_7_HOME=/usr/lib/jvm/java-7-openjdk-amd64
JAVA_8_HOME=/usr/lib/jvm/java-8-openjdk-amd64

function runBM() {
  ALGEBIRD_VERSION=$1
  SCALA_VERSION=$2
  JAVA_VER=$3
  export JAVA_HOME=/usr/lib/jvm/java-${JAVA_VER}-openjdk-amd64
  cd ~/workspace/github/$ALGEBIRD_VERSION
  LOG_OUTPUT_PATH=$OUTPUT_FOLDER/${ALGEBIRD_VERSION}_JAVA-${JAVA_VER}_$SCALA_VERSION.log
  OUTPUT_PATH=$OUTPUT_FOLDER/${ALGEBIRD_VERSION}_JAVA-${JAVA_VER}_$SCALA_VERSION.csv
  ./sbt ++$SCALA_VERSION clean algebird-benchmark/"run -i 20 -wi 15 -f1 -p numElements=10000 -p depthK=6 -t1 -o $LOG_OUTPUT_PATH -rf csv .*QTreeBenchmark.*"
  cp algebird-benchmark/jmh-result.csv $OUTPUT_PATH
}

JAVA_VERSIONS="8"
SCALA_VERSIONS="2.10.5 2.11.7"
ALGEBIRD_VERSIONS="algebird algebird2"
for jvm in $JAVA_VERSIONS; do
  for scala_ver in $SCALA_VERSIONS; do
    for alge in $ALGEBIRD_VERSIONS; do
      runBM $alge $scala_ver $jvm
    done
  done
done
```

**Comment by johnynek** [comment]
merge when green

**Pull request merged**